packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/preprocessor/Preprocessor.java

```diff
@@ -87,14 +85,8 @@ public CodeModel processCodeModel() {
 
         CodeModel codeModel;
         try {
-            if (!file.startsWith("{")) {
-                // YAML
-                codeModel = yamlMapper.loadAs(file, CodeModel.class);
-            } else {
-                try (JsonReader jsonReader = JsonProviders.createReader(file)) {
-                    codeModel = CodeModel.fromJson(jsonReader);
-                }
-            }
+            // YAML
+            codeModel = yamlMapper.loadAs(file, CodeModel.class);
```

Comment:
Nice catch. I don't recall we ever reached the `else` part. 
@weidongxu-microsoft for confirmation as well.

Reply:
Yeah, I don't think the `else` would have effect. Likely an ancient legacy.